### DISCUSSION TITLE
randomized strings/favs not showing in my test site for a game im making

### DISCUSSION THREAD

**Discussion by DevRenamon-Alder** [community/community#139557] [Programming Help]
### Body

I'm trying to add a randomized favorite food, toy type, and specific toy using random numbers and variables for each of the 5 different digimon click menus. these are added to the html of the popup menus through getting the html span elements by their id and changing their inner HTML to the variables assigned the random string values. It stopped working after i changed the stuff that makes the popups visible to also check if the digimon1 through 6's values are false and hides the popup despite being clicked if they are (due to that implying a lack of data to show). I thought this was completely unrelated so there is a chance its not the cause of the favorite items not showing up. heres a link to the repo.  https://github.com/DevRenamon-Alder/Test-PartneredDigimon

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by steffinjack4** [comment]
It is like you're on the right track with how you're assigning the random values. I had a similar issue once and it turned out the variables were being set after the DOM update, so the span stayed empty [Sweetgreen](https://sweetgreenmenu.us/). Maybe check the order your functions run when the popup is triggered.
